src/app/features/date-range/services/controller/date-range.service.ts

```diff
@@ -33,8 +38,22 @@ export class DateRangeService {
       .afterClosed()
       .pipe(
         switchMap((period: UnavailableDates | null) => {
-          return period ? of(period) : of(null);
+          return this.addPersonalInfo(period);
```

Comment:
this service should not think about such stuff. Otherwise it is not universal any more. Till exempel we are going to use it in admin module. This case it is not matter has user personal data eller not. this should be moved to controller.

Reply:
done